operator/gitops/argocd/pipeline-service/pipelines-as-code/allow-argocd.yaml

```diff
@@ -1,42 +1,39 @@
 ---
-apiVersion: rbac.authorization.k8s.io/v1
-kind: RoleBinding
-metadata:
-  name: allow-argo-cd-rolebinding
-  namespace: pipelines-as-code
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: ClusterRole
-  name: admin
-subjects:
-  - kind: ServiceAccount
-    name: openshift-gitops-argocd-application-controller
-    namespace: openshift-gitops
----
 kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
-  name: crd-manager-for-build
+  name: openshift-gitops-pipelines-as-code
 rules:
-  - verbs:
+  - apiGroups:
+    - apiextensions.k8s.io
+    resources:
+      - customresourcedefinitions
+    verbs:
       - patch
       - get
       - list
       - create
-      - get
-    apiGroups:
-      - apiextensions.k8s.io
+  - apiGroups:
+      - admissionregistration.k8s.io
     resources:
-      - customresourcedefinitions
+      - validatingwebhookconfigurations
+    verbs:
+      - create
+  - apiGroups:
+      - monitoring.coreos.com
+    resources:
+      - servicemonitors
```

Comment:
Why are we providing permission to ArgoCD for collecting metrics? Could you help me understand where its collecting and displaying those metrics?

Reply:
pipelines-as-code [requires](https://github.com/openshift-pipelines/pipelines-as-code/releases/download/v0.14.2/release.yaml) the creation of a servicemonitor, so we are providing argoCD with the permissions to create that.